libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts

```diff
@@ -364,15 +364,14 @@ export class TelemetryLoggerMiddleware implements Middleware {
 
     private populateAdditionalChannelProperties(activity: Activity, properties?: { [key: string]: string }): void {
         if (activity) {
+            const channelData = activity.channelData;
             switch (activity.channelId) {
                 case 'msteams':
-                    const channelData = activity.channelData as TeamsChannelData;
-
-                    properties['TeamsTenantId'] = channelData.tenant ? channelData.tenant.id : '';
                     properties['TeamsUserAadObjectId'] = activity.from ? activity.from.aadObjectId : '';
 
-                    if (channelData.team) {
-                        properties['TeamsTeamInfo'] = JSON.stringify(channelData.team);
+                    if (channelData as TeamsChannelData) {
```

Comment:
I recently added a type guard for this, which you're welcome to use (https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/teamsActivityHelpers.ts#L11).

To me, it's more semantically correct than casting with `as`. This also compiles away to just a truthy check whereas the type guard at least verifies that `channelData` is an object.

The type guard could be expanded to check types of existing properties but I just kept it simple for now.

Reply:
Yeah, it's a bit annoying having to write type guards for larger objects. That said, it's a huge dev-exp win both for SDK devs and external devs.